core/src/main/java/eu/maveniverse/maven/njord/shared/SessionConfig.java

```diff
@@ -587,10 +623,58 @@ public Optional<String> prefix() {
                 return Optional.ofNullable(prefix);
             }
 
+            @Override
+            public Map<String, Map<String, String>> serverConfigurations() {
+                return serverConfigurations;
+            }
+
+            @Override
+            public Optional<Map<String, String>> serverConfiguration(String id) {
+                return Optional.ofNullable(serverConfigurations.get(id));
+            }
+
             @Override
             public Optional<CurrentProject> currentProject() {
                 return Optional.ofNullable(currentProject);
             }
         }
     }
+
+    static Map<String, Map<String, String>> extractServerConfigurations(Map<String, Object> configProperties) {
+        HashMap<String, Map<String, String>> serverConfigurations = new HashMap<>();
+        configProperties.entrySet().stream()
+                .filter(e -> e.getKey().startsWith("aether.transport.wagon.config.")
```

Comment:
why is every key prefixed with `aether.transport.wagon.config.` or `aether.connector.wagon.config.`? I think this deserves some comment pointing to https://github.com/apache/maven/blob/9b95526182ed2be5e72b191308896aebf0ff73b2/impl/maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java#L241. I couldn't find the 2nd prefix (is it used in Maven3)?